lib/plug_datadog_stats/cowboy.ex

```diff
@@ -0,0 +1,52 @@
+defmodule PlugDatadogStats.Cowboy do
+  @moduledoc """
+  Cowboy-specific HTTP request metrics not covered by the more
+  general base PlugDatadogStats module.
+  """
+
+  @doc """
+  onresponse callback for integrating with Cowboy.
+  To use, configure Plug to set this function as an
+  onresponse handler for Cowboy, in config.exs:
+  http: [
+    protocol_options: [
+      onresponse: &PlugDatadogStats.Cowboy.onresponse/4,
+    ]
+  ]
+  """
+  def onresponse(status, _headers, _body, request) do
+    # Cowboy-induced HTTP 400 detection logic borrowed from Plug.
+    if status == 400 and empty_headers?(request) do
```

Comment:
is it guaranteed that the headers will be empty in this case?

Reply:
Yes, this logic is 100% "borrowed" from Plug. See https://github.com/elixir-plug/plug/commit/4d4c48e0bb5e4ae2b6fd9f2d031c8b7eb40ce290#diff-618de3f6bf97fa90fa716e798031b884R240.

Reply:
I believe the idea is that Cowboy doesn't even bother processing or including the headers if they're invalid.